### PR TITLE
Add longPathAware manifest to NuGet.Build.Tasks.Console

### LINKED ISSUE #12121: [Bug]: Make static restore graph evaluation long path aware
### NuGet Product Used

dotnet.exe, MSBuild.exe

### Product Version

all versions since static restore added

### Worked before?

_No response_

### Impact

It's more difficult to complete my work

### Repro Steps & Context

Use the NUGET_PACKAGES environment variable, or `globalPackagesFolder` config in a `nuget.config` file to set the GPF to a path of around 200 characters long. Then, restore a project with `msbuild -t:restore -p:RestoreUseStaticGraphEvaluation=true` from a Developer PowerShell. Make sure the registry/group policy setting for long paths is enabled, and this restore will fail. But if you remove `-p:RestoreUseStaticGraphEvaluation=true` from the arguments, it will work.

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/12121

Regression? No

## Description

Copied https://github.com/dotnet/roslyn/blob/ee9808b0ec1d5b94dba1ade21319b61487588ced/src/Compilers/Core/Portable/Resources/default.win32manifest

then added `windowsSettings` and `longPathAware` as per blog post: https://learn.microsoft.com/en-us/archive/blogs/jeremykuhne/net-4-6-2-and-long-paths-on-windows-10

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception: I tried writing a test that would extract the win32 manifest and look for the `longPathAware` element, but it turned out to be too difficult. Similarly, to write a test to validate the app can read/write long paths would require starting it as a new process, which means generating valid input files, which again is in my opinion not worth the effort.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by erdembayar — APPROVED** [review]
I assume you already manually tested it. If you include any screenshot before and after change would be great.

File: src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <assemblyIdentity version="1.0.0.0" name="MyApplication.app"/>
```
Comment (jeffkl) [inline src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest:3]: Will this cause any issues not having the correct assembly information?  Do we need any more settings like MSBuild.exe has?

https://github.com/dotnet/msbuild/blob/main/src/MSBuild/MSBuild.exe.manifest

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest:3]: We probably don't need that at all.

We have a manifest for nuget.exe, maybe we should be matching that, https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/app.manifest.

Comment (zivkan) [reply src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest:3]: > Will this cause any issues not having the correct assembly information?

If you extract the app manifest of any .NET Framework app that does not specify its own app.manifest, then, as stated in the PR description, it gets roslyn's [default app manifest](https://github.com/dotnet/roslyn/blob/ee9808b0ec1d5b94dba1ade21319b61487588ced/src/Compilers/Core/Portable/Resources/default.win32manifest). The app manifest is unchanged from this default, including calling the assembly `MyApplication.app`.

> Do we need any more settings like MSBuild.exe has?

In Visual Studio, when you use the new item wizard to add an app manifest, it contains all the `supportedOS` compatibility, and possibly one or two other sections, but I question it's usefulness given that Roslyn's default doesn't contain it, and nobody has complained about compatibility issues yet. It also doesn't specify Windows 11. Does that mean none of these apps are compatible with Windows 11? 🤷

> We have a manifest for nuget.exe, maybe we should be matching that,

I considered that, but as hinted at in my previous paragraph, many of the settings are not in the default manifest when you build a .NET Framework app without a manifest. I don't know what all the other settings do, what unintended side effects they'll have, so I feel more comfortable using the default settings, and adding the one we know we want.

But if we as a team prefer to do things differently, we can.

**Review by jeffkl — APPROVED** [review]
:shipit:

**Review by dominoFire — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest (hunk 1 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest:3]: Following an internal discussion, we appear to be satisfied with the fact that it seems to work, despite the non-obvious name mismatch. FWIW, [the win32 assembly identity](https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#assemblyidentity) does not appear to have any relation to .NET assembly identities. It's unfortunate that they both call themselves assemblies, but they're not referring to the same thing.

**Pull request merged**